app/bundles/LeadBundle/Controller/LeadController.php

```diff
@@ -305,12 +305,9 @@ public function viewAction(Request $request, IntegrationHelper $integrationHelpe
         /** @var \Mautic\LeadBundle\Model\LeadModel $model */
         $model = $this->getModel('lead.lead');
 
-        // When we change company data these changes get cached
-        // so we need to clear the entity manager
-        $model->getRepository()->clear();
-
         /** @var \Mautic\LeadBundle\Entity\Lead $lead */
         $lead = $model->getEntity($objectId);
+        $model->getRepository()->refetchEntity($lead);
```

Comment:
Detaching instead of clearing doesn't do the same job?

Reply:
There is no entity to detach